cql3/statements/create_keyspace_statement.cc

```diff
@@ -175,6 +176,30 @@ std::optional<sstring> check_restricted_replication_strategy(
             break;
         }
     }
+    // The minimum_keyspace_rf configuration option can be used to forbid
+    // a lower replication factor. We assume that all numeric replication
+    // options are replication factors - this is true for SimpleStrategy and
+    // NetworkTopologyStrategy but in the future if we add more strategies,
+    // we may need to limit this test only to specific options.
+    // A zero replication factor is not forbidden - it is the traditional
+    // way to avoid replication on some DC.
+    // We ignore errors (non-number, negative number, etc.) here,
```

Comment:
The try-catch blocks is exactly to ignore errors if somebody sets the RF to some non-numeric junk - because we already have another place in the code that checks this. The only purpose of *this* code is to check, if RF is actually a number, that is not a forbidden number.